src/main/java/com/example/labj1/logic/OrderLogic.java

```diff
@@ -27,26 +24,30 @@ public class OrderLogic {
 
 
     @Transactional
-    public void createOrder(Order order) {
+    public ResponseEntity<?> createOrder(Order order) {
         System.out.println("order: " + order);
         String inventoryUrl = "http://localhost:8081/inventory/v2";
 
         UriComponentsBuilder url = UriComponentsBuilder.fromUriString(inventoryUrl)
 
                 .queryParam("productId", (order.getLineItems().get(0).getProductInfo().getId()));
-        // .queryParam("productId", 24234);
-
         HttpHeaders headers = new HttpHeaders();
         headers.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON));
 
         HttpEntity<String> entity = new HttpEntity<>(headers);
         Inventory inventory = restTemplate.exchange(url.build().toUri().toString(), HttpMethod.GET, entity, Inventory.class).getBody();
```

Comment:
Move this out to a different package (maybe /dependencies, /microservices, or /downstreamapis) and class called `InventoryDependency`, `InventoryService`, or something like that. The reason for that is separation of concerns, but it is also so that other places in your code can make this call if they needed to. You are in essence creating an API client for dealing with inventory domain communications/actions that can be shared across your application.